compiler/test/suites/optimizations.re

```diff
@@ -211,79 +211,125 @@ describe("optimizations", ({test}) => {
   );
   assertAnf(
     "test_local_mutations1",
-    "let mut x = 5; x = 6",
+    "export let foo = () => {let mut x = 5; x = 6}",
```

Comment:
Why was this test changed?

Reply:
Top-level mutables are now just regular wasm globals, so the optimization doesn't do anything at the top level anymore. By pushing this down into the body of a function, it makes sure that the optimization is exercised!

Reply:
Makes sense! I wanted to make sure that this change wasn't making this breaking, but it doesn't seem like it does (just different)